plugins/ccgx/fu-ccgx-firmware.c

```diff
@@ -287,6 +296,60 @@ fu_ccgx_firmware_parse_md_block(FuCcgxFirmware *self, FwupdInstallFlags flags, G
 	return TRUE;
 }
 
+typedef struct {
+	FuCcgxFirmware *self;
+	FwupdInstallFlags flags;
+} FuCcgxFirmwareTokenHelper;
+
+static gboolean
+fu_ccgx_firmware_tokenize_cb(GString *token, guint token_idx, gpointer user_data, GError **error)
+{
+	FuCcgxFirmwareTokenHelper *helper = (FuCcgxFirmwareTokenHelper *)user_data;
+	FuCcgxFirmware *self = FU_CCGX_FIRMWARE(helper->self);
+
+	/* remove WIN32 line endings */
+	g_strdelimit(token->str, "\r\x1a", '\0');
```

Comment:
isn't this only going to effectively remove EOF?

Reply:
Nah, it's per-token, i.e. operating on the allocated `line`, not the file.

Reply:
Then shouldn't it be looking for just carriage return not EOF delimiter?

Reply:
Isn't the last token going to include the EOF?

Reply:
oh right - in that case the srec plugin probably needs to match this too.

Reply:
Indeed, fixed, good catch.